torchaudio/functional.py

```diff
@@ -1194,6 +1194,52 @@ def contrast(
     return output_waveform
 
 
+def dcshift(
+        waveform: Tensor,
+        shift: float,
+        limiter_gain: float = None
```

Comment:
To accomodate `None`, the type signature should be something like 
```suggestion
        limiter_gain: Optional[float] = None
```

with 

```python
from typing import Optional
```

Reply:
torch script test is giving below error when using Optional[float] = None
```
RuntimeError: 
Could not cast value of type Optional[float] to bool:
'dcshift' is being compiled since it was called from 'func'
```

Reply:
Oops, torch script is fussy about optional.
Can you try this?

```python
    if limiter_gain is not None:
        limiter_threshold = 1.0 - (abs(shift) - limiter_gain)
    ...

    if limiter_gain is not None and shift > 0:
    ...
    elif limiter_gain is not None and shift < 0:
```

If this works, you can get rid of `use_limiter` variable.

Reply:
@mthrok . When using ```limiter_gain: float = None``` torch  script test is giving below error
```
Expected a default value of type float on parameter "limiter_gain":
```
is there another way to use None or should i revert back to float comparison?

Reply:
Oh, sorry, I meant applying both changes. `Optional[float]` and `limiter_gain is not None`. I tested it just now and the test passes.


```diff
@@ -1197,7 +1198,7 @@ def contrast(
 def dcshift(
         waveform: Tensor,
         shift: float,
-        limiter_gain: float = None
+        limiter_gain: Optional[float] = None
 ) -> Tensor:
     r"""Apply a DC shift to the audio. Similar to SoX implementation.
     This can be useful to remove a DC offset
@@ -1218,18 +1219,16 @@ def dcshift(
     """
     output_waveform = waveform
     limiter_threshold = 0.
-    use_limiter = False

-    if limiter_gain:
-        use_limiter = True
+    if limiter_gain is not None:
         limiter_threshold = 1.0 - (abs(shift) - limiter_gain)

-    if use_limiter and shift > 0:
+    if limiter_gain is not None and shift > 0:
         mask = waveform > limiter_threshold
         temp = (waveform[mask] - limiter_threshold) * limiter_gain / (1 - limiter_threshold)
         output_waveform[mask] = (temp + limiter_threshold + shift).clamp(max=limiter_threshold)
         output_waveform[~mask] = (waveform[~mask] + shift).clamp(min=-1, max=1)
-    elif use_limiter and shift < 0:
+    elif limiter_gain is not None and shift < 0:
         mask = waveform < -limiter_threshold
         temp = (waveform[mask] + limiter_threshold) * limiter_gain / (1 - limiter_threshold)
         output_waveform[mask] = (temp - limiter_threshold + shift).clamp(min=-limiter_threshold)
(PY3.8-cuda101) moto@044c5eb8233e:/scratch/moto/torchaudio$ git checkout torchaudio/functional.py
```

Reply:
@mthrok . Now it is working .Thanks